### ISSUE TITLE
[BUG] temporal fusion transformer trained with GPU's then loaded with map_locations=torch.device('cpu') does not apply the correct device to loss metric

### ISSUE THREAD

**Issue by arizzuto** [sktime/pytorch-forecasting#1753]
**Describe the bug**

**To Reproduce**

```python
<Paste your code here>
```

**Expected behavior**

**Additional context**

**Versions**
<details>

</details>
